### ISSUE TITLE
[BUG] Massive RAM usage and crashing after 1-2 hours

### ISSUE THREAD

**Issue by Radiicall** [Moosync/Moosync#236]
**Describe the bug**
Moosync seems to be increasing in ram usage over time and then it crashes after a while.

**To Reproduce**

1. Open moosync
2. use it
3. ram usage creeps up
4. crash

- OS: Fedora Linux 38
- Version: Flatpak v8.1.1

**Additional context**
`$ journalctl -xe`
```
May 14 01:03:28 shklung app.moosync.moosync.desktop[202057]: <--- Last few GCs --->
May 14 01:03:28 shklung app.moosync.moosync.desktop[202057]: [3:0x37c1b2dc000]      212 ms: Scavenge 3.0 (4.6) -> 2.6 (4.8) MB, 1.67 / 0.00 ms  (average mu = 1.000, current mu = 1.000) task;
May 14 01:03:28 shklung app.moosync.moosync.desktop[202057]: <--- JS stacktrace --->
May 14 01:03:28 shklung app.moosync.moosync.desktop[202057]: FATAL ERROR: NewSpace::EnsureCurrentCapacity Allocation failed - JavaScript heap out of memory
May 14 01:04:37 shklung app.moosync.moosync.desktop[202056]: /app/bin/moosync: line 2:     3 Aborted                 (core dumped) /app/Moosync/moosync --no-sandbox "$@"
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]: node:events:491
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:       throw er; // Unhandled 'error' event
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:       ^
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]: Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at new NodeError (node:internal/errors:399:5)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at target.send (node:internal/child_process:740:16)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at Object.sendEvent (/app/Moosync/resources/app.asar/spotify.js:1:7669)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at EventEmitter.emit (node:events:513:28)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at _positionHolder.callback (/app/Moosync/resources/app.asar/node_modules/librespot-node/dist/index.js:1:2630)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at Timeout.<anonymous> (/app/Moosync/resources/app.asar/node_modules/librespot-node/dist/index.js:1:7705)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at listOnTimeout (node:internal/timers:569:17)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at process.processTimers (node:internal/timers:512:7)
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]: Emitted 'error' event on process instance at:
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at node:internal/child_process:744:35
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:     at process.processTicksAndRejections (node:internal/process/task_queues:77:11) {
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]:   code: 'ERR_IPC_CHANNEL_CLOSED'
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]: }
May 14 01:04:37 shklung app.moosync.moosync.desktop[202170]: Node.js v18.14.0
May 14 01:04:37 shklung systemd[2219]: app-flatpak-app.moosync.moosync-202039.scope: Consumed 8min 48.732s CPU time.
```

Program starts at around 200-400mb but slowly but surely creeps up, I took a screenshot of it reaching 2.9GB and still increasing
![image](https://github.com/Moosync/Moosync/assets/66682497/1f78ef5b-a639-4b2f-8418-09d5d2424bdd)

**Comment by Nemo0312** [comment]
Random crashes can be related to the scanning interval. In the settings under paths> Scan Intervals, set that value to -1.

**Comment by Ovenoboyo** [comment]
Yup this seems to be an issue with the scanner

**Comment by Radiicall** [comment]
> Yup this seems to be an issue with the scanner

I'll try setting the scan interval to -1 and check back with the results later today

**Comment by Radiicall** [comment]
RAM usage seems to not be increasing anymore, this option should probably be disabled by default until its fixed or removed all together

**Comment by Ovenoboyo** [comment]
Hi, sorry for the delay but this should be fixed in v9.0.0+. I'm closing this issue since it seems to be fixed, feel free to open it if you feel otherwise :)
